truffleruby/src/main/ruby/core/enumerable.rb

```diff
@@ -36,11 +36,11 @@ def chunk(&original_block)
     ::Enumerator.new do |yielder|
       previous = nil
       accumulate = []
-      block = if initial_state.nil?
-        original_block
+      if initial_state.nil?
+        block = original_block
       else
         duplicated_initial_state = initial_state.dup
-        Proc.new{ |val| original_block.yield(val, duplicated_initial_state)}
+        block = Proc.new{ |val| original_block.yield(val, duplicated_initial_state)}
```

Comment:
Rubocop enforces this transformation? Personally I find using if as an expression quite useful. Having 2 assignments instead of just one increases complexity of the code.